src/fusor/fusion_matching.py

```diff
@@ -149,115 +281,107 @@ def _match_fusion_structure(
             possible genes element
         :param is_five_prime_partner: If the 5' fusion partner is being compared
-        :return: A score indiciating the degree of match or None if the partner is
-            ? or v
+        :param mi: A MatchInformation object
         """
-        # Set default match score
-        match_score = 0
-
         # If the assayed partner is unknown or the categorical partner is a multiple
         # possible gene element, return None as no precise information
         # regarding the compared elements is known
         if isinstance(assayed_element, UnknownGeneElement) or isinstance(
             categorical_element, MultiplePossibleGenesElement
         ):
-            return None
+            return
 
         # Compare gene partners first
         if assayed_element.gene == categorical_element.gene:
-            match_score += 1
-        else:
-            return 0
+            if is_five_prime_partner:
+                mi.five_prime_gene = True
+            else:
+                mi.three_prime_gene = True
 
         # Then compare transcript partners if transcript data exists
         if isinstance(assayed_element, TranscriptSegmentElement) and isinstance(
             categorical_element, TranscriptSegmentElement
         ):
             if assayed_element.transcript == categorical_element.transcript:
-                match_score += 1
-            else:
-                return 0
+                if is_five_prime_partner:
+                    mi.five_prime_transcript = True
+                else:
+                    mi.three_prime_transcript = True
 
             start_or_end = "End" if is_five_prime_partner else "Start"
             fields_to_compare = [
-                f"exon{start_or_end}",
-                f"exon{start_or_end}Offset",
-                f"elementGenomic{start_or_end}",
+                ("exon", f"exon{start_or_end}"),
+                ("exon_offset", f"exon{start_or_end}Offset"),
+                ("breakpoint", f"elementGenomic{start_or_end}"),
             ]
 
             # Determine if exon number, offset, and genomic breakpoint match
-            for field in fields_to_compare:
-                if getattr(assayed_element, field) == getattr(
-                    categorical_element, field
+            for mi_field, element_field in fields_to_compare:
+                if getattr(assayed_element, element_field) == getattr(
+                    categorical_element, element_field
                 ):
-                    match_score += 1
-                else:
-                    return 0
-
-        return match_score
+                    if is_five_prime_partner:
+                        setattr(mi, f"five_prime_{mi_field}", True)
+                    else:
+                        setattr(mi, f"three_prime_{mi_field}", True)
 
     def _compare_fusion(
         self, assayed_fusion: AssayedFusion, categorical_fusion: CategoricalFusion
```

Comment:
Probably for a follow-up: Would it be possible for this to accept `AssayedFusion|CategoricalFusion` to do the matches?

Reply:
In theory, yes. Would need to refactor some of the helper methods to allow for AssayedFusions too. I would probably omit ITDs from this however.